app/controllers/api/v3/providers_controller.rb

```diff
@@ -7,7 +7,9 @@ def index
         build_fields_for_index
 
         @providers = if params[:search].present?
-                       @recruitment_cycle.providers.search_by_code_or_name(params[:search])
+                       return render(status: :bad_request) if params[:search].length < 2
+
+                       @recruitment_cycle.providers.search_by_code_or_name(params[:search]).limit(5)
```

Comment:
why was a limit of 5 added?

Reply:
Ah - that should not be there. I'll remove.